File: src/pt_frontend/src/consts/entities.ts
import { createCandidVariant } from '@/utils/create-candid-variant';

export const ENTITY = {
  DOCUMENT: 'Document',
  ORGANIZATION: 'Organization',
  PROJECT: 'Project',
  REVISION: 'Revision',
  ROLE: 'Role',
  USER: 'User',
  INVITE: 'Invite',
  WORKFLOW: 'Workflow',
  LOG: 'LogEntry',
} as const;

const entityNames = Object.values(ENTITY);

export const entity = createCandidVariant(entityNames);

File: src/shared/src/types/entities.rs
use candid::CandidType;
use serde::Deserialize;

#[derive(CandidType, Deserialize, Clone, Debug, PartialEq)]
pub enum Entity {
    User,
    Document,
    Revision,
    Organization,
    Invite,
    Project,
    Workflow,
    LogEntry,
}
